apps/api/src/domain/interfaces/id-manager.interface.ts

```diff
@@ -0,0 +1,24 @@
+import { NotAvailable } from '../../lib/types/not-available';
+import { NotFound } from '../../lib/types/not-found';
+import { OK } from '../../lib/types/ok';
+import { AggregateId } from '../types/AggregateId';
+import { IIdGenerator } from './id-generator.interface';
+
+export interface IIdManager extends IIdGenerator {
```

Comment:
When do we use just the IIdGenerator interface by itself?

Reply:
We currently use this in non-create commands in order to generate UUIDs. However, we need to also `use` the ID after we `generate`.  We should fix this when working on [this story](https://www.pivotaltracker.com/story/show/182539181).